backend/fms_core/template_importer/row_handlers/normalization/normalization.py

```diff
@@ -0,0 +1,100 @@
+        if destination_sample['concentration_nm'] is None and destination_sample['concentration_ngul'] is None:
+            self.errors['concentration'] = 'A concentration in either nM or ng/uL must be specified.'
+
+        if destination_sample['concentration_nm'] is not None and destination_sample['concentration_ngul'] is not None:
+            self.errors['concentration'] = 'Concentration must be specified in either nM or ng/uL, not both.'
+
+        source_sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=source_sample['container']['barcode'],
+            coordinates=source_sample['coordinates'])
+
+        destination_container_dict = destination_sample['container']
+
+        parent_barcode = destination_container_dict['parent_barcode']
+        if parent_barcode:
+            container_parent_obj, self.errors['parent_container'], self.warnings['parent_container'] = get_container(
+                barcode=parent_barcode)
+        else:
+            container_parent_obj = None
+
+        if source_sample_obj and (container_parent_obj or not parent_barcode) and "concentration" not in self.errors.keys():
+            concentration = None
+            # Case when ng/uL is given
+            if destination_sample['concentration_ngul']:
+                concentration = destination_sample['concentration_ngul']
+            # Case when nM is given
+            elif destination_sample['concentration_nm']:
+                if source_sample_obj.is_library:
+                    library = source_sample_obj.derived_sample_not_pool.library
+                    if library.library_size:
+                        concentration = convert_concentration_from_nm_to_ngbyul(destination_sample['concentration_nm'],
+                                                                                library.molecular_weight_approx,
+                                                                                library.library_size)
+                    else:
+                        self.errors['library'] = 'Library size has not been set for this library.'
+                    if concentration is None:
+                        self.errors['concentration'] = 'Failed to convert concentration from nM to ng/uL'
+                else:
+                    self.errors['concentration'] = 'Concentration specified in nM should be only for libraries.'
+
+            is_concentration_valid, self.errors['concentration_validation'], self.warnings['concentration_validation'] \
+                = validate_normalization(initial_volume=process_measurement['volume_used'],
+                                         initial_concentration=source_sample_obj.concentration,
+                                         final_volume=destination_sample['volume'],
+                                         desired_concentration=concentration,
+                                         tolerance=1)
```

Comment:
You can leave it to default here we'll tweak the default to be a good compromise for normalization and if we need to reuse we'll use the param to get it right if the default is not good enough.

Reply:
After testing 1 is much too permissive... you should use the default of 0.01, we could possibly go lower ... but it may get annoying. Our calculation at the third decimal should always pass at the 0,01 level and they will prevent gross errors at least.